server/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthRequest.java

```diff
@@ -229,17 +203,17 @@ public ClusterHealthRequest(StreamInput in) throws IOException {
             waitForEvents = Priority.readFrom(in);
         }
         waitForNoInitializingShards = in.readBoolean();
-        if (in.getVersion().onOrAfter(Version.V_4_3_0)) {
-            indicesOptions = IndicesOptions.readIndicesOptions(in);
-        } else {
-            indicesOptions = IndicesOptions.LENIENT_EXPAND_OPEN;
+        if (in.getVersion().onOrAfter(Version.V_4_3_0) && in.getVersion().before(Version.V_6_0_0)) {
```

Comment:
Given that nodes from < 5 can't join a 6.x cluster anymore I think we could remove the 4.3.0 condition.